Remove deprecated code from `wcs`

eerovaher opened the pull request:
### Description

The first commit removes code deprecated in #10229. The users are meant to import things from `astropy.wcs.wcsapi` and apparently the internals of that module are considered private. The deprecation therefore had no change log entry, and the removal of the code shouldn't have an entry either.

The second commit removes code deprecated in #6682. There were no change log entries announcing the deprecation, so likewise there's no entry announcing the removal.

There is one more deprecated function:
https://github.com/astropy/astropy/blob/98d82dd9477622fd53aab6ed84efb03b6147674c/astropy/wcs/wcs.py#L966-L972
But in this case I'm not sure if removing the deprecated function is the right thing to do. Maybe the warning type should be changed instead. I'd prefer if someone more familiar with `wcs` would have a look at this.

<!-- Optional opt-out -->

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

pllim commented on astropy/wcs/wcs.py:
```diff
@@ -255,14 +254,6 @@ def __init__(
-                f"Function received unexpected arguments ({list(kwargs)}) these "
-                "are ignored but will raise an Exception in the "
-                "future.",
-                AstropyDeprecationWarning,
```
Can we be sure it actually throws an exception now?

eerovaher replied:
Python will now raise a `TypeError`, we don't have to implement anything ourselves.

pllim commented:
Still need a change log.

eerovaher commented:
I'd first like to know what should be done with `_read_d2im_old_format()`.

pllim commented:
This? Yeah, not sure. Looks like the warning is always emitted if this gets called. Maybe @mcara would know. But this can also be a follow-up PR because I feel like bigger changes are needed to get rid of this warning.

```python
    def _read_d2im_old_format(self, header, fobj, axiscorr):
        warnings.warn(
            "The use of ``AXISCORR`` for D2IM correction has been"
            " deprecated.`~astropy.wcs` will read in files with ``AXISCORR`` but"
            " ``to_fits()`` will write out files without it.",
            AstropyDeprecationWarning,
        )
```

eerovaher commented:
To elaborate: from the deprecation warning in `_read_d2im_old_format()` it is not clear to me that _reading_ files with AXISCORR is or even should be deprecated, so it's not obvious to me that this function should be removed. Should it be the case that the warning type needs to be changed, but doing that is not trivial, then it would indeed be a good idea to leave it for a dedicated pull request, but I'd still like that to be decided here.

eerovaher commented:
I haven't heard anything about `_read_d2im_old_format()`, but that shouldn't prevent the removal of other deprecated code. As far as I can tell the file that actually defines `SlicedLowLevelWCS` is not considered to be part of the public API and there was no change log entry for deprecating the old file, so I haven't added a change log entry for its removal either. For `NoConvergence` `**kwargs` I added a change log entry like @pllim requested. I also rebased the pull request to pick up the RTD configuration update from yesterday.

pllim reviewed (approved):
LGTM now. Thanks!
